### PR TITLE
Changing the IssuerReference adding CertificateType and updating sample

### PR DESCRIPTION
**Pull request by pomortaz**
1. Changing the IssuerReference to IssuerParameters
2. Adding CertificateType 
3. Updating conditions on listing count
4. Moving samples to a sample file and simplify the readme samples

### CODE REVIEW

**Comment by pomortaz** [comment]
@rohmano and @kunalcha please review.

File: lib/services/keyVault/lib/keyVault.js [hunk 1]
```diff
@@ -327,9 +327,10 @@ KeyVaultClient.prototype.getKey = function (keyIdentifier, options, callback) {
  * @param {string} keyName The name of the key
  * 
  * @param {object} [options] Optional Parameters.
- * 
- * @param {number} [options.maxresults] Maximum number of results to return.
- * 
+ *
+ * @param {number} [options.maxresults] Maximum number of results to return in
+ * a page. If not specified the service will return up to 25 results.
```
Comment (kunalcha) [inline lib/services/keyVault/lib/keyVault.js]: >25 [](start = 58, length = 2)

If service changes this value, does this documentation become outdated?

File: lib/services/keyVault/lib/keyVaultClient.js [hunk 2]
```diff
@@ -3666,6 +3700,16 @@ KeyVaultClient.prototype.getSecretVersions = function (vaultBaseUrl, secretName,
     if (maxresults !== null && maxresults !== undefined && typeof maxresults !== 'number') {
       throw new Error('maxresults must be of type number.');
     }
+    if (maxresults !== null && maxresults !== undefined) {
+      if (maxresults > 25)
```
Comment (kunalcha) [inline lib/services/keyVault/lib/keyVaultClient.js]: >25 [](start = 23, length = 2)

Do we need flexibility at service side to change this value over time?

File: lib/services/keyVault/lib/keyVault.js (hunk 1 above)
Comment (amarzavery) [inline lib/services/keyVault/lib/keyVault.js]: then you need to update the swagger spec and regenerate the clients. Basically follow the pipeline.

File: lib/services/keyVault/lib/keyVaultClient.js (hunk 2 above)
Comment (pomortaz) [inline lib/services/keyVault/lib/keyVaultClient.js]: For now this is the limitation as stated by REST API and this issue: https://github.com/Azure/azure-sdk-for-node/issues/1842
I haven't seen any ask for change in this value.

**Comment by kunalcha** [comment]
:shipit:

**Comment by pomortaz** [comment]
@amarzavery can you please merge?

**Pull request merged**